File: articles/nudge-theory-behavioral-public-policy/fortran/nudge_welfare.f90
program nudge_welfare
  implicit none

  integer :: adopted
  real :: utility, user_benefit, social_benefit
  real :: friction_cost, admin_cost, implementation_cost, welfare

  utility = 0.65
  adopted = 1
  user_benefit = 0.50 * adopted
  social_benefit = 0.40 * adopted
  friction_cost = 0.06
  admin_cost = 0.05
  implementation_cost = 0.073

  welfare = utility + user_benefit + social_benefit - friction_cost - admin_cost - implementation_cost

  print *, "Synthetic nudge policy welfare:", welfare
end program nudge_welfare

File: articles/nudge-theory-behavioral-public-policy/rust/nudge_welfare.rs
fn total_welfare(
    utility: f64,
    adopted: f64,
    friction_cost: f64,
    admin_cost: f64,
    implementation_cost: f64,
) -> f64 {
    let user_benefit = 0.50 * adopted;
    let social_benefit = 0.40 * adopted;

    utility + user_benefit + social_benefit - friction_cost - admin_cost - implementation_cost
}

fn main() {
    let w = total_welfare(0.65, 1.0, 0.06, 0.05, 0.073);
    println!("Synthetic nudge policy welfare: {:.3}", w);
}
